js/chips.js

```diff
@@ -19,67 +25,112 @@ function selectChip(chip) {
 }
 
-function squareContains(selectedChip) {
-	// decide if a valid square is empty.
-	// if not empty, decide if the square contains a chip
-	// if the square contains a chip, find out if it is an enemy chip!
-	// isEnemyChip(selectedChip, evalChip)
-	// return possible values: "emptySquare", "friendlyChip", "enemyChip"
+function evalSquare(selectedSquare) {
+
+	// if the clicked square is the parent of the selectedChip, do nothing.
+	if (selectedSquare.children[0] == selectedChip) {
+		return false; 
+	}
+
+	var squareHas;
+
+	if (selectedSquare.children.length === 0) {
+		squareHas = "emptySquare";
+	} else if (isEnemyChip(selectedSquare.children[0])) {
+		squareHas = "enemyChip";
+	} else {
+		squareHas = "friendlyChip";
+	}
+	
+	return squareHas;
+
 }
 
-function isEnemyChip(selectedChip, evalChip) {
-	// check the selectedChip's color and test it against another chip, `evalChip` to decide if it is an enemy chip.
-	// return a boolean
+function isEnemyChip(evalChip) {
+	if (!selectedChip) {
+		return false;
+	}
+
+	var chipColor;
+	var evalChipColor;
+	
+	if (selectedChip.classList.contains("red")) {
+		chipColor = "red";
+	} else {
+		chipColor = "black";
+	}
+
+	if (evalChip.classList.contains("red")) {
+		evalChipColor = "red";
+	} else {
+		evalChipColor = "black";
+	}
+	
+	if (chipColor === evalChipColor) {
+		return false;
+	} else {
+		return true;
+	}
 }
 
 function userClick(element) {
 	//takes a parameter of e.target, any clicked element on the page
 
+	// if the clicked element is also the selectedChip, do nothing.
+	if (element == selectedChip) {
+		// this condition will execute first and the return false here will ensure that the remainder 
+		// of the userClick() function does not execute.
+		return false; 
+	}
+
 	//if element is chip...
 	if (element.classList.contains('chip')) {
-
-		//check if chip already selected
-		if (element.classList.contains('selected')) {
-				clearSelection();
+			//select that chip
+			selectChip(element);
 
 			//check if it's your own chip, NOT an enemyChip
 			// else if (!isEnemyChip(element, evalChip))
 			//this step commented out for testing purposes. Game will select any chip.
-		} else {
-			//select that chip
-			selectChip(element);
-		}
+		
 		//if element is square
-	} else if (element.classList.contains('square')) {
+	} else if (element.classList.contains('red-square')) {
+			console.log("that's a red square!");
+			
 		//if square is within an array of valid moves for the last .selected chip!
 		//then make move: update state/location of chip, run the right animation
 		//else
 		//then perhaps alert user: "not a valid move"?
 		//if element is not a square, nor a chip = it's outside the board, clear the selection
+	} else if (element.classList.contains('black-square')) {
+		var squareHas = evalSquare(element);
+		console.log("that's a black square!");
+		console.log("squareHas = "+squareHas);
```

Comment:
here is where I am really evaluating the square and its' chip content (or lack thereof!) if you checkout this branch and load the checkers-flex.html file in your browser with the developer console open, you will see lots of console logs!